### PR TITLE
SPEC-1587 Update Windows stapled OCSP test plan

### PR DESCRIPTION
**Pull request by vincentkam**
https://jira.mongodb.org/browse/SPEC-1587

See also: https://github.com/mongodb/specifications/pull/767

### CODE REVIEW

File: .evergreen/orchestration/configs/servers/windows/rsa-basic-tls-ocsp-mustStaple.json [hunk 1]
```diff
@@ -0,0 +1 @@
+../rsa-basic-tls-ocsp-mustStaple.json
```
Comment (vincentkam) [inline .evergreen/orchestration/configs/servers/windows/rsa-basic-tls-ocsp-mustStaple.json]: Symlink, since the files are currently identical

Comment (ShaneHarvey) [reply .evergreen/orchestration/configs/servers/windows/rsa-basic-tls-ocsp-mustStaple.json]: Does this work? I remember symlinks causing problems in windows a few years ago.

Comment (vincentkam) [reply .evergreen/orchestration/configs/servers/windows/rsa-basic-tls-ocsp-mustStaple.json]: Hm a good question! I'd forgotten this runs in Cygwin on the Windows hosts. Symlinks are supposed to work on Windows with Git on Windows 10 in dev mode, but for simplicity's sake, I'll use a copy.

File: .evergreen/orchestration/configs/servers/windows/README.md [hunk 2]
```diff
@@ -2,10 +2,14 @@ These "disableStapling" configuration files have the OCSP failpoint
 removed because the failpoint results in the mongod erroring out on
 Windows.
```
Comment (kevinAlbs) [inline .evergreen/orchestration/configs/servers/windows/README.md]: Hah I missed that this directory existed. Are these separate configs necessary? Stapling is not supported in macOS either, but I think the failpoint is just a no-op in Windows/macOS based on the comments in the server code:
https://github.com/mongodb/mongo/search?q=disablestapling&unscoped_q=disablestapling
Testing on my Windows machine with  mongod v4.5.0-138-gac9caed I'm able to start it with that failpoint:
`mongod.exe --setParameter enableTestCommands=1 --setParameter failpoint.disableStapling="{'mode': 'alwaysOn'}"`

Additionally, my WIP for windows testing is able to at least start mongod on Windows with the other configs (and fails later for other reasons). Here's one that starts with rsa-basic-tls-ocsp-mustStaple-disableStapling.json: https://evergreen.mongodb.com/task_log_raw/mongo_c_driver_ocsp_ocsp_winssl_malicious_server_test_1_rsa_delegate_patch_d6ced5f6a02fef6be59eba09917f1b788e3d5bec_5e6984d83e8e86487c4dda6f_20_03_12_00_40_23/0?type=T#L412

So perhaps we can remove this directory entirely?

Comment (vincentkam) [reply .evergreen/orchestration/configs/servers/windows/README.md]: Oh nice! It didn't always use to be a no-op. You're right we can remove this directory!

**Pull request merged**